src/test/base32_tests.cpp

```diff
@@ -31,9 +32,45 @@ BOOST_AUTO_TEST_CASE(base32_testvectors)
 
     // Decoding strings with embedded NUL characters should fail
     BOOST_CHECK(!DecodeBase32("invalid\0"s)); // correct size, invalid due to \0
-    BOOST_CHECK(DecodeBase32("AWSX3VPP"s)); // valid
+    BOOST_CHECK( DecodeBase32("AWSX3VPP"s)); // valid
     BOOST_CHECK(!DecodeBase32("AWSX3VPP\0invalid"s)); // correct size, invalid due to \0
     BOOST_CHECK(!DecodeBase32("AWSX3VPPinvalid"s)); // invalid size
 }
 
+BOOST_AUTO_TEST_CASE(base32_padding)
+{
+    // Is valid without padding
+    {
+        const std::vector<uint8_t> in{'f', 'o', 'o', 'b', 'a'};
+        const std::string out{"mzxw6ytb"};
+        BOOST_CHECK_EQUAL(EncodeBase32(in), out);
+    }
+
+    // Valid size
+    BOOST_CHECK(!DecodeBase32("========"s));
+    BOOST_CHECK(!DecodeBase32("a======="s));
+    BOOST_CHECK( DecodeBase32("aa======"s));
+    BOOST_CHECK(!DecodeBase32("aaa====="s));
+    BOOST_CHECK( DecodeBase32("aaaa===="s));
+    BOOST_CHECK( DecodeBase32("aaaaa==="s));
+    BOOST_CHECK(!DecodeBase32("aaaaaa=="s));
+    BOOST_CHECK( DecodeBase32("aaaaaaa="s));
+}
+
+BOOST_AUTO_TEST_CASE(base32_roundtrip) {
```

Comment:
Shouldn't this be a fuzzing test?

Reply:
I've modelled these based on https://github.com/bitcoin/bitcoin/blob/master/src/test/base58_tests.cpp#L84-L98 - but I guess the existing fuzz tests already cover this as well, so I'll migrate them all to https://github.com/bitcoin/bitcoin/blob/master/src/test/fuzz/base_encode_decode.cpp#L20-L54, thanks for the hint

Reply:
Done